src/Pagination/Pagination.tsx

```diff
@@ -23,37 +24,60 @@ export interface PaginationProps {
 }
 
 export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((props: PaginationProps): React.ReactElement<void> => {
-    const Initialoffset: number = props.offset ? props.offset : 10;
+
+    const [list, setList] = React.useState<Array<number>>([]);
+    const [dotnavList, setDotnavList] = React.useState<Array<number>>([]);
+    const [pagingSize, setPagingSize] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        const initialOffset: number = props.offset ? props.offset : 10;
+        setPagingSize(Math.ceil(props.size / initialOffset));
+    }, [props.offset, props.pagingLength, props.size]);
+
+    React.useEffect(() => {
+        const intialLength: number = props.pagingLength ? props.pagingLength : 5;
+        getList(props.value, intialLength);
+    }, [pagingSize, props.value]);
 
     /**
      * Generates an array of the pages that needs to be displayed
      * It depends on the size, offset, and the current value
      * @param {number} value The current value. The current page.
-     * @param {number} size The size of the pagination
-     * @param {number} offset The offset to show the numbers
+     * @param {number} size The size of the pagination or length of items
+     * @param {number} offset The number of items to show per page
      * @returns {Array<number>} An array of the pages that needs to be displayed
      */
-    function getList(value: number, size: number, offset: number): Array<number> {
-        const list: Array<number> = [];
-        const range: { min: number, max: number } = { min: 0, max: 0 };
-        if (value <= (Math.ceil(offset / 2))) {
-            for (let i = 1; i <= offset; i++) {
-                list.push(i);
-            }
-        } else if (value >= (size + 1 - (Math.ceil(offset / 2)))) {
-            range.min = size + 1 - offset;
-            range.max = size;
-            for (let i = range.min; i <= range.max; i++) {
-                list.push(i);
-            }
-        } else {
-            range.min = value + 1 - (Math.ceil(offset / 2));
-            range.max = range.min + offset - 1;
-            for (let i = range.min; i <= range.max; i++) {
-                list.push(i);
+    function getList(value: number, length: number): void {
```

Comment:
Don't you think a method called `getList` should return something? if not, then it shouldn't be called `getList` just to not confuse the reader